### PR TITLE
User Profile - Rename access to labels for request, response and mappings

### PR DESCRIPTION
**Pull request by ywangd**
It is agreed that the "access" field should be renamed to the more
generic "labels". Data stored in this field are meant to be application
specific (similar to "data" but searchable) and ES does not attempt to
understand its content or purpose. Hence the more generic name is a
better fit.

### CODE REVIEW

File: x-pack/docs/en/rest-api/security/activate-user-profile.asciidoc [hunk 1]
```diff
@@ -43,7 +43,7 @@ is intended for.
 
 `access_token`::
 (Required*, string)
-The user's access token. If you specify the `access_token` grant type, this
+The user's labels token. If you specify the `access_token` grant type, this
```
Comment (n1v0lg) [inline x-pack/docs/en/rest-api/security/activate-user-profile.asciidoc:46]: Think this should still be `access`

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java [hunk 2]
```diff
@@ -1403,7 +1403,7 @@ LogEntryBuilder withRequestBody(UpdateProfileDataRequest updateProfileDataReques
             XContentBuilder builder = JsonXContent.contentBuilder().humanReadable(true);
             builder.startObject()
                 .field("uid", updateProfileDataRequest.getUid())
-                .field("access", updateProfileDataRequest.getAccess())
+                .field("access", updateProfileDataRequest.getLabels())
```
Comment (n1v0lg) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java:1406]: Should this be "labels"?

**Review by n1v0lg — APPROVED** [review]
LGTM! Two possible typos. Build failure looks unrelated.

File: x-pack/docs/en/rest-api/security/activate-user-profile.asciidoc (hunk 1 above)
Comment (ywangd) [reply x-pack/docs/en/rest-api/security/activate-user-profile.asciidoc:46]: You are absolutely right! Thanks

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java (hunk 2 above)
Comment (ywangd) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java:1406]: Yes. Good catch. Thanks!

**Comment by ywangd** [comment]
@elasticmachine run elasticsearch-ci/part-1-fips

**Comment by ywangd** [comment]
@elasticmachine update branch

**Comment by ywangd** [comment]
FIPS failure is this issue https://github.com/elastic/elasticsearch/issues/85803

**Comment by ywangd** [comment]
@elasticmachine run elasticsearch-ci/part-1

**Comment by ywangd** [comment]
@elasticmachine update branch

**Review by albertzaharovits — APPROVED** [review]
LGTM

**Review by sethmlarson — APPROVED** [review]

**Pull request merged**